### PR TITLE
Support for before message and after message

### PR DESCRIPTION
**Pull request by rohitharkhani**
Added support for the after message and before message handling. 
Not sure about the test cases i dint found anything existing for the afterrequest and afterresponse.

### CODE REVIEW

File: lib/engine_socketio.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ SocketIoEngine.prototype.step = function (requestSpec, ee, opts) {
           context.socketio.on(response.channel, function receive(data) {
             done = true;
             context.socketio.off(response.channel);
-            let functionNames = _.concat(opts.aftermessage || [], requestSpec.emit.response.aftermessage || []);
+            let functionNames = _.concat(opts.afterMessage || [], requestSpec.emit.response.afterMessage || []);
```
Comment (hassy) [inline lib/engine_socketio.js]: 👍

File: lib/engine_socketio.js [hunk 2]
```diff
@@ -83,93 +86,131 @@ SocketIoEngine.prototype.step = function (requestSpec, ee) {
...
+                    return next(err);
+                  }
+                  return next(null);
+                });
+              },
+              function done(err) {
+                if (err) {
+                  return callback(err, context);
+                }
+                err = processResponse(ee, data, response.data);
```
Comment (hassy) [inline lib/engine_socketio.js]: This needs to be called asynchronously (but should be fixed by the rebase)

**Comment by hassy** [comment]
A test would be a good as well - you can look at a similar test case for HTTP in [the main repo](https://github.com/shoreditch-ops/artillery/blob/master/test/test.sh#L117).

**Comment by rohitharkhani** [comment]
@hassy Thanks for review i will pull updates from repository and change accordingly. 

afterMessage added in response because for socket.io we need channel to listen response, So without response afterMessage is useless. Now if we will move it out, while writing test plan if some had specified afterMessage but not specified response then afterMessage will not executed and may be it will create confusion why it is not executed. To encapsulate complete response handling i have added it to response. 

Please share your view i will update according to that one.

**Comment by rohitharkhani** [comment]
Here we might need to think about the scenario level hooks and should we prompt it on any message?

**Pull request closed without merging**